docs/function.rst

```diff
@@ -0,0 +1,147 @@
+.. _function:
+
+Function
+========
+
+Melody functions are user-written Python functions that perform the
```

Comment:
I'm slightly concerned that just calling them 'functions' is confusion. Maybe something as simple as strictly calling them "Melody Functions" will suffice. On the other hand, in the world of optimisation is it really a Target Function?